### PR TITLE
Feat: Newsletter form and useNewsletter hook

### PR DESCRIPTION
**Pull request by lariciamota**
## What's the purpose of this pull request?

Adds a newsletter form and a hook that uses the [subscribeToNewsletter mutation](https://github.com/vtex/faststore/pull/1385).

## How does it work?

It uses the `subscribeToNewsletter` mutation to add the person's info to the store's newsletter list.

## How to test it?

The Newsletter form is in the footer, fill in the form fields and then check if it was saved on MasterData using this GET:
https://storeframework.vtexcommercestable.com.br/api/dataentities/NL/search?email=HERE
- Substitute _HERE_ with the email filled in the form
- Add a cookie VtexIdclientAutCookie with your auth token

## References

<em>Spread the knowledge: is this any content you used to create this PR that is worth sharing?</em>

<em>Extra tip: add references to related issues or mention people important to this PR may be good for the documentation and reviewing process</em>

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**
- [ ] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [ ] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 

**PR Description**
- [ ] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [ ] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [ ] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.

**Documentation**
- [ ] PR description
- [ ] Added to/Updated the Storybook - *if applicable*.
- [ ] For documentation changes, ping @ carolinamenezes, @ PedroAntunesCosta or @ Mariana-Caetano to review and update.

### CODE REVIEW

File: src/components/sections/Newsletter/Newsletter.tsx [hunk 1]
```diff
@@ -12,22 +14,22 @@ export interface NewsletterProps
...
-  /**
-   * Callback function when submitted.
-   */
-  onSubmit: (value: string) => void
 }
 
 const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
-  function Newsletter({ title, subtitle, onSubmit, ...otherProps }, ref) {
+  function Newsletter({ title, subtitle, ...otherProps }, ref) {
+    const { addUser, loading } = useNewsletter()
```
Comment (victorhmp) [inline src/components/sections/Newsletter/Newsletter.tsx:21]: Should we also be handling the `success` or `fail` case? Like showing a `"Subscribed to the newsletter!"` or `"Could not subscribe"` message after the mutation has been completed?

File: src/sdk/newsletter/useNewsletter.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  id: string
+}
+
+export const mutation = gql`
+  mutation AddToNewsletter($data: IPersonNewsletter!) {
+    addToNewsletter(data: $data) {
+      id
+    }
+  }
+`
```
Comment (victorhmp) [inline src/sdk/newsletter/useNewsletter.ts:20]: [Nitpick ⛏️ ]
```suggestion
`

```

**Review by victorhmp — APPROVED** [review]
LGTM!
But I really wish we didn't have to handle the request state "manually" :(

The `useNewsletter` hook would become sooo much simpler.

**Review by eduardoformiga — COMMENTED** [review]
Good job!

While I was reviewing I remember that we have [this PR](https://github.com/vtex-sites/base.store/pull/242) with the markup and also this [Figma prototype](https://www.figma.com/file/zHMOtOPduiM0hNNcg21iR1/Handoff-v1(PDP)?node-id=36%3A41903). Do you know if we have plans to implement it?

File: src/components/common/Footer/Footer.tsx [hunk 3]
```diff
@@ -16,6 +17,7 @@ function Footer() {
   return (
     <footer className="footer layout__content-full">
       <IncentivesFooter />
+      <Newsletter title="Newsletter" />
```
[the remark this answers is not in the document]
Comment (lariciamota) [reply src/components/common/Footer/Footer.tsx]: It makes a lot of sense! I moved it there

File: src/components/sections/Newsletter/Newsletter.tsx (hunk 1 above)
Comment (lariciamota) [reply src/components/sections/Newsletter/Newsletter.tsx]: Yeah, I think we should, and we could do that using the "data" and "error" variables. Although, we could do it in another PR once we have a design decision on how the newsletter should be displayed in our example stores. What do you think?

**Comment by lariciamota** [comment]
> Good job!
> 
> While I was reviewing I remember that we have [this PR](https://github.com/vtex-sites/base.store/pull/242) with the markup and also this [Figma prototype](https://www.figma.com/file/zHMOtOPduiM0hNNcg21iR1/Handoff-v1(PDP)?node-id=36%3A41903). Do you know if we have plans to implement it?

Yeah, the Newsletter file used in this current PR is the same one created on the PR you sent! 
About the Figma prototype: I don't have access to it :/ but we really should put the improvement of the Newsletter's look on our backlog!

File: src/components/sections/Newsletter/Newsletter.tsx [hunk 4]
```diff
@@ -44,17 +51,28 @@ const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
           </div>
 
           <div data-newsletter-controls>
```
Comment (eduardoformiga) [inline src/components/sections/Newsletter/Newsletter.tsx:53]:
```suggestion
          <div data-fs-newsletter-controls>
```

File: src/components/sections/Newsletter/Newsletter.tsx [hunk 5]
```diff
@@ -44,17 +51,28 @@ const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
...
+              ref={nameInputRef}
+              required
+            />
             <Label htmlFor="newsletter-email">Your email</Label>
             <Input
               id="newsletter-email"
               type="email"
               name="newsletter-email"
               ref={emailInputRef}
+              required
```
Comment (eduardoformiga) [inline src/components/sections/Newsletter/Newsletter.tsx:68]: Nice!

File: src/components/sections/Newsletter/Newsletter.tsx [hunk 6]
```diff
@@ -12,26 +16,29 @@ export interface NewsletterProps
...
-        onSubmit(emailInputRef.current?.value ?? '')
-      }
+      formElement.reset()
     }
 
     return (
-      <section data-store-newsletter>
+      <Section data-store-newsletter>
         <Form
           data-newsletter-form
```
Comment (eduardoformiga) [inline src/components/sections/Newsletter/Newsletter.tsx:43]:
```suggestion
          data-fs-newsletter-form
```

line 48: `data-fs-newsletter-header` as well. Github didn't give me the chance to suggest haha

**Review by eduardoformiga — APPROVED** [review]
Nice addition! 👏 

I've tried to give you access to the [Figma prototype](https://www.figma.com/file/zHMOtOPduiM0hNNcg21iR1/Handoff-v1(PDP)?node-id=36%3A41903) aforementioned now. I also agree that we can deal with the design specs on another task in our backlog.

@renatamottam do we have a task for `Newsletter` refinement? I didn't find it here, so we can create 😉

File: src/components/sections/Newsletter/Newsletter.tsx [hunk 7]
```diff
@@ -12,26 +16,29 @@ export interface NewsletterProps
...
 
-      if (emailInputRef.current?.value !== '') {
-        onSubmit(emailInputRef.current?.value ?? '')
-      }
+      formElement.reset()
     }
 
     return (
-      <section data-store-newsletter>
+      <Section data-store-newsletter>
```
[the remark this answers is not in the document]
Comment (lariciamota) [reply src/components/sections/Newsletter/Newsletter.tsx:41]: Yes, it is! I'll update them

**Review by victorhmp — APPROVED** [review]

**Pull request merged**